contracts/governance/extensions/GovernorSequentialProposalId.sol

```diff
@@ -0,0 +1,32 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+import {Governor} from "../Governor.sol";
+
+abstract contract GovernorSequentialProposalId is Governor {
+    error NextProposalIdCanOnlyBeSetOnce();
+
+    uint256 private _numberOfProposals;
+    mapping(uint256 proposalHash => uint256 proposalId) private _proposalIds;
+
+    function _getProposalId(
+        address[] memory targets,
+        uint256[] memory values,
+        bytes[] memory calldatas,
+        bytes32 descriptionHash
+    ) internal virtual override returns (uint256) {
+        uint256 proposalHash = super._getProposalId(targets, values, calldatas, descriptionHash);
+
+        uint256 storedProposalId = _proposalIds[proposalHash];
+        return storedProposalId == 0 ? (_proposalIds[proposalHash] = ++_numberOfProposals) : storedProposalId;
```

Comment:
My idea was that this function should be a view getter, it shouldn't automatically generate a new proposal id, and it should revert if there is no proposal id stored (because the proposal was never created).